File: src/main/java/io/fairspace/callisto/model/Collection.java
package io.fairspace.callisto.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.ZonedDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder(toBuilder = true)
public class Collection {
    private Long id;
    private String type;
    private String location;
    private String name;
    private String description;
    private String uri;
    private Access access;
    ZonedDateTime dateCreated;
    String creator;
}

File: src/main/kotlin/io/fairspace/ceres/events/model/Collection.kt
package io.fairspace.ceres.events.model

import java.net.URI

data class Collection(
    val id: Long = 0,
    val type: String = "",
    val location: String = "",
    val name: String = "",
    val description: String? = null,
    val uri: String = "",
    val access: String = "None"
)
